nuxeo-features/nuxeo-platform-task/nuxeo-platform-task-core/src/main/java/org/nuxeo/ecm/platform/task/core/service/TaskServiceImpl.java

```diff
@@ -192,13 +198,20 @@ public String endTask(CoreSession coreSession, NuxeoPrincipal principal, Task ta
 
     @Override
     public boolean canEndTask(NuxeoPrincipal principal, Task task) {
-        if (task != null && (!task.isCancelled() && !task.hasEnded())) {
+        if (task != null) {
```

Comment:
Here you're changing the behavior of a public API, you can call `isTaskActive` before checking user assignation.

Reply:
If I understood correctly, if I call `isTaskActive` here and I change the order as the comment below, the code won't reach ConcurrentUpdateException.
I believe we should still call `isTaskActive` to keep the current behaviour, but not sure about the suggested changed below.

Reply:
Hum, you're right. I think we have no choice to have something like `checkPermission(Task, String/NuxeoPrincipal)` and change `canEndTask` to something like:
```
public boolean canEndTask(NuxeoPrincipal principal, Task task) {
    return isTaskActive(task) && checkPermission(task, principal);
```
Wdyt?